### PR TITLE
Tensorflow, tensorboard, tensorboard-data-server

### PR DESCRIPTION
**Pull request by dav-dap**
###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [X ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by doronbehar** [comment]
Please also split the updates to 1 commit per update, as explained in [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md?rgh-link-date=2023-07-12T12%3A25%3A02Z)

File: pkgs/development/python-modules/tensorflow/default.nix [hunk 1]
```diff
@@ -98,7 +98,7 @@ let
 
   tfFeature = x: if x then "1" else "0";
 
-  version = "2.11.1";
```
[the remark this answers is not in the document]
Comment (doronbehar) [reply pkgs/development/python-modules/tensorflow/default.nix:101]: > Just tried, but it does not output the right hash code:
> 
> ```
> error: invalid SRI hash 'AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA'
> ```
> 
> with no other info

Try to use exactly this fake hash: `sha256-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=` or, use `lib.fakeHash`.

**Comment by doronbehar** [comment]
BTW I think since this update may include more non trivial changes, I'll cc @jyp as well.

File: pkgs/development/python-modules/tensorflow/default.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (dav-dap) [reply pkgs/development/python-modules/tensorflow/default.nix:101]: > Try to use exactly this fake hash: `sha256-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=`

That solves the problem with the hash, but I am experiencing problems with bazel:
```
  /build/source/third_party/repo.bzl:89:35: in <toplevel>
WARNING: Download from https://storage.googleapis.com/mirror.tensorflow.org/github.com/google/benchmark/archive/f7547e29ccaed7b64ef4f7495ecfff1c9f6f3d03.tar.gz failed: class com.google.devtools.build.lib.bazel.repository.downloader.UnrecoverableHttpException Unknown host: storage.googleapis.com
WARNING: Download from https://github.com/google/benchmark/archive/f7547e29ccaed7b64ef4f7495ecfff1c9f6f3d03.tar.gz failed: class com.google.devtools.build.lib.bazel.repository.downloader.UnrecoverableHttpException Unknown host: github.com
ERROR: An error occurred during the fetch of repository 'com_google_benchmark':
   Traceback (most recent call last):
        File "/build/source/third_party/repo.bzl", line 73, column 33, in _tf_http_archive_impl
                ctx.download_and_extract(
Error in download_and_extract: java.io.IOException: Error downloading [https://storage.googleapis.com/mirror.tensorflow.org/github.com/google/benchmark/archive/f7547e29ccaed7b64ef4f7495ecfff1c9f6f3d03.tar.gz, https://github.com/google/benchmark/archive/f7547e29ccaed7b64ef4f7495ecfff1c9f6f3d03.tar.gz] to /build/output/external/com_google_benchmark/temp16390992179402712944/f7547e29ccaed7b64ef4f7495ecfff1c9f6f3d03.tar.gz: Unknown host: github.com
ERROR: /build/source/WORKSPACE:15:14: fetching _tf_http_archive rule //external:com_google_benchmark: Traceback (most recent call last):
        File "/build/source/third_party/repo.bzl", line 73, column 33, in _tf_http_archive_impl
                ctx.download_and_extract(
Error in download_and_extract: java.io.IOException: Error downloading [https://storage.googleapis.com/mirror.tensorflow.org/github.com/google/benchmark/archive/f7547e29ccaed7b64ef4f7495ecfff1c9f6f3d03.tar.gz, https://github.com/google/benchmark/archive/f7547e29ccaed7b64ef4f7495ecfff1c9f6f3d03.tar.gz] to /build/output/external/com_google_benchmark/temp16390992179402712944/f7547e29ccaed7b64ef4f7495ecfff1c9f6f3d03.tar.gz: Unknown host: github.com
ERROR: no such package '@com_google_benchmark//bazel': java.io.IOException: Error downloading [https://storage.googleapis.com/mirror.tensorflow.org/github.com/google/benchmark/archive/f7547e29ccaed7b64ef4f7495ecfff1c9f6f3d03.tar.gz, https://github.com/google/benchmark/archive/f7547e29ccaed7b64ef4f7495ecfff1c9f6f3d03.tar.gz] to /build/output/external/com_google_benchmark/temp16390992179402712944/f7547e29ccaed7b64ef4f7495ecfff1c9f6f3d03.tar.gz: Unknown host: github.com
INFO: Elapsed time: 2.114s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (0 packages loaded)
FAILED: Build did NOT complete successfully (0 packages loaded)

error: builder for '/nix/store/jm8ajbkmzj2qi3xhh3pjb2srp684qxyb-tensorflow-2.13.0.drv' failed with exit code 1
error: 1 dependencies of derivation '/nix/store/zdm6wfqas80c6397cb9d7s9gfl5cslq4-python3.10-tensorflow-2.13.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/922vl5a33vdlv1q8y8afmm81bxlld18g-python3-3.10.12-env.drv' failed to build
error: 1 dependencies of derivation '/nix/store/hx5l3nl28njyrwvzajgl976d5wq6vrsq-system-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/rkrz2irf6r1gdxvldps76bhjhpkhzpsj-nixos-system-nixos-22.11.1069.cbe419ed4c8.drv' failed to build
```

[the remark this answers is not in the document]
Comment (doronbehar) [reply pkgs/development/python-modules/tensorflow/default.nix:101]: > That solves the problem with the hash, but I am experiencing problems with bazel:

This is a pretty hard package you chose to update as a 1st time contribution to Nixpkgs! Please push your updated hash changes here, and rewrite the Git commit history such that each update is on it's own commit, with commit titles as described in [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md?rgh-link-date=2023-07-12T12%3A28%3A51Z). Hopefully the maintainers will be able to help at some point. I'm sorry I can't help any further - I'm clueless as for anything bazel related.

[the remark this answers is not in the document]
Comment (dav-dap) [reply pkgs/development/python-modules/tensorflow/default.nix:101]: I do not have much choice, that's the package I need.
I rewrote the history.
Now, on the issue above - it seems it is trying to downlod from a couple of links that do not exist. Does anyone know how to get to the lines generating that instruction? A simple `grep -r` with the commit code across the whole `nixpkgs` does not return anything

File: pkgs/development/python-modules/tensorboard/default.nix [hunk 2]
```diff
@@ -24,15 +24,14 @@
 
 buildPythonPackage rec {
   pname = "tensorboard";
-  version = "2.11.0";
+  version = "2.13.0";
   format = "wheel";
-  disabled = pythonOlder "3.6" || pythonAtLeast "3.11";
```
Comment (doronbehar) [inline pkgs/development/python-modules/tensorboard/default.nix:29]: @dav-dap this line was changed in https://github.com/NixOS/nixpkgs/pull/242660 , and now this PR has merge conflicts.

**Review by doronbehar — COMMENTED** [review]
Moreover, it seems the maintainers of these packages are not responsive enough to coordinate the different changes these packages get. So I'd advice to close this PR in favor of https://github.com/NixOS/nixpkgs/pull/227957 which seems like it had progressed much more.

**Comment by doronbehar** [comment]
Oh and BTW for next time, the commit messages should be:

```
{pkg-name}: {old version} -> {new version}
```

**Comment by dav-dap** [comment]
Closing in order to follow https://github.com/NixOS/nixpkgs/pull/227957 which is further than this current work

**Comment by dav-dap** [comment]
Closing in order to follow the more advanced https://github.com/NixOS/nixpkgs/pull/227957

**Pull request closed without merging**